apps/dataset-browser/src/app/[locale]/page.tsx

```diff
@@ -19,27 +38,53 @@ const filterKeysOrder: ReadonlyArray<keyof SearchResult['filters']> = [
   'licenses',
 ];
 
+export interface FacetMenuProps {
+  filters: SearchResult['filters'];
+  filterKeysOrder: ReadonlyArray<keyof SearchResult['filters']>;
+}
+
+function FacetMenu({filterKeysOrder, filters}: FacetMenuProps) {
+  const t = useTranslations('Home');
+
+  return (
+    <>
+      <SearchField />
+      {filterKeysOrder.map(
+        filterKey =>
+          !!filters[filterKey]?.length && (
+            <FilterSet
+              key={filterKey}
+              title={t(`${filterKey}Filter`)}
+              searchResultFilters={filters[filterKey]}
+              filterKey={filterKey}
+              testId={`${filterKey}Filter`}
+            />
+          )
+      )}
+    </>
+  );
+}
+
 interface Props {
   searchParams?: SearchParams;
 }
 
-// Revalidate the page
-export const revalidate = 0;
-
 export default async function Home({searchParams}: Props) {
   const searchOptions = fromSearchParamsToSearchOptions(searchParams ?? {});
   const sortBy = getClientSortBy({
     sortBy: searchOptions.sortBy,
     sortOrder: searchOptions.sortOrder,
   });
 
-  let hasError, searchResult;
+  let hasError, searchResultOrUndefined;
   try {
-    searchResult = await datasetFetcher.search(searchOptions);
+    searchResultOrUndefined = await datasetFetcher.search(searchOptions);
   } catch (error) {
     hasError = true;
     console.error(error);
   }
+  const searchResult = searchResultOrUndefined as SearchResult;
```

Comment:
To me it looks a bit strange to convert a `searchResultOrUndefined` to a `SearchResult` - as if you're ignoring the `undefined`. Is this conversion necessary?

Reply:
I had some weird type errors within the map on line 165. Outside the map Typescript understood that the type of `searchResult` was `SearchResult`. But within the map, the type of `searchResult` was `any`. I changed the code a bit, so I don't need the `searchResultOrUndefined` variable.